CDCgov/prime-reportstream
# Kotlin Data Types to Typescript Codegen Plugin

This PR adds a new kotlin project to the repo that creates a gradle plugin for exporting kotlin data types to typescript. This plugin will take into account classes/fields/methods annotated with JsonIgnore/JsonIgnoreType. It adds this plugin to the prime-router project with some initial pending classes selected for export.  Typescript definitions are written to frontend-react/typings/api-codegen.d.ts. The plugin supplies a new gradle task called generateTypescriptDefinitions that is set to run as a dependency of the classes lifecycle task. The expected workflow would be that any changes to kotlin data structures that are being exported to the frontend will be automatically converted to typescript when rebuilt. As those types begin to be used on the frontend, any breaking changes to those structures will cause the frontend to fail to build.

Also includes minor changes to better enforce prime-router's kotlin version to transitive packages and eager task mitigation.

Test Steps:
1. Find a suitable kotlin data class that is used for the azure API and either apply the @TsExport annotation to it, or manually add its fully qualified name to the manualClass list in  prime-router's build.gradle.kts
2. Run the build task
3. Verify that the generateTypescriptDefinitions task runs in the console output
4. Verify that the data class and all related types have been added to the api-codegen.d.ts file

## Changes
- tsGeneratePlugin project added
- prime-router includes tsGeneratePlugin via includeBuild in settings.gradle.kts
- tsGeneratePlugin configured for usage on prime-router
- some initial kotlin classes have been configured for export

## Checklist

### Testing

### Process
